### ISSUE TITLE
Request for adding Warning/Error feature to Transformer layer when dropout set to 1.0

### ISSUE THREAD

**Issue by kitrak-rev** [pytorch/pytorch#96633]
### 🚀 The feature, motivation and pitch

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">The model is not &quot;turned off during training&quot;. With dropout=1.0, for dropout layers you&#39;ll get all zero at train and, apparently, identity at test. I don&#39;t think pytorch should have allowed dropout=1.0. It should be ValueError, not sure I get the reasoning there.</p>&mdash; Andrej Karpathy (@karpathy) <a href="https://twitter.com/karpathy/status/1635062921398206464?ref_src=twsrc%5Etfw">March 12, 2023</a></blockquote> 

As suggested by Andrej Karpathy, it makes sense to raise an error or atleast a warning when the user sets dropout in the transformer layer.  So it would be great if someone can decide whether it should raise an error or warning , and which warning or error to raise in the above case.

P.S :  I would like to be **assigned** the task for **implementation** of this issue.
